config/pgoconfig.go

```diff
@@ -477,6 +478,13 @@ func (c *PgoConfig) Validate() error {
 
 	if c.Cluster.User == "" {
 		return errors.New(errPrefix + "Cluster.User is required")
+	} else {
+		//checks that username is valid
+		re := regexp.MustCompile("^[a-z0-9.-]*$")
+		if !re.MatchString(c.Cluster.User) {
```

Comment:
Let's use the Kubernetes validation around this for DNS labels based on RFC1035:

https://godoc.org/k8s.io/apimachinery/pkg/util/validation#IsDNS1035Label

Reply:
Per further discussion and research with @jmckulk it turns out for secrets we want this function: https://godoc.org/k8s.io/apimachinery/pkg/util/validation#IsConfigMapKey